### PR TITLE
I am going into battle, and I only want your strongest potions

### PR DESCRIPTION
**Pull request by MMMiracles**
My potions would kill you, traveler. You can't handle my potions.

Adds healing, poison, flame, kinetic, and speed potions to the game. Meant to be more of something that could be found as adventure loot or potential addition to the wizard's arsenal. 

Flasks are left unlabeled because there's nothing more fun than trying your luck on an unmarked potion.

### CODE REVIEW

**Comment by KorPhaeron** [comment]
I'm sorry but SS13 is a science fiction game I'm not sure this fits the theme

File: code/modules/reagents/chemistry/reagents/potion_reagents.dm [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	id = "poisonpotion"
+	description = "A strange mixture of various components. Foul fumes drift off the horrible mixture as it occassionally bubbles. Why would anyone make this?"
+	color = "#00FF00" // rgb: 0, 255, 0
+
+/datum/reagent/consumable/potion/poison/reaction_mob(mob/living/M, method=INGEST, reac_volume, show_message = 1)
+	if(iscarbon(M) && M.stat != DEAD)
+		if(method in list(TOUCH, VAPOR, PATCH))
+			if(show_message)
+				M << "<span class='userdanger'>A horrible pain washes over you, it feels like something is eating away at your skin!</span>"
+				M.adjustBruteLoss(25)
```
Comment (coiax) [inline code/modules/reagents/chemistry/reagents/potion_reagents.dm]: This is weird. Wouldn't you just hit the target with 1u, wait for it to decay, then hit them again?

Comment (MMMiracles) [inline code/modules/reagents/chemistry/reagents/potion_reagents.dm]: whoops, ill make it based on amount used

**Comment by Dorsisdwarf** [comment]
I agree that Majjikal Potions don't fit retro-future Space Scifi

**Comment by KazeEspada** [comment]
Wizards mang

**Comment by PKPenguin321** [comment]
>I agree that Majjikal Potions don't fit retro-future Space Scifi

time to remove chemistry then Kappa

**Comment by MMMiracles** [comment]
idea is for more filler when it comes to ruin loot past guns/ore/death.

also i'll probably look into making a ruin with a mysterious hermit who you can exchange gold bars to for a random potion.

**Comment by Dorsisdwarf** [comment]
>Mixing space chemicals the same as  _literal_ magic potion flasks handed out by hermits for gold.

**Comment by MMMiracles** [comment]
got a problem with hermits?

**Comment by Core0verload** [comment]
No. Turn it into actual chem mixes that can be repilcated by players.

**Comment by MMMiracles** [comment]
@Core0verload they're not meant for normal mixing, they're meant as loot items or potential wizard gear.

**Comment by Core0verload** [comment]
Chemists can make stronger mixes anyway.

**Comment by coiax** [comment]
If you can make potion bottles for them that smash minecraft style into smoke when you throw them, then I think you'll have rounded out the features nicely.

**Comment by MMMiracles** [comment]
I could probably add a throw_impact to spawn smoke with the the potion reagent.

 I made the bottles a child of the liquor bottles so you could throw them at people or smash them on the head with it.

**Comment by optimumtact** [comment]
They're not magic Kor, they're chemical mixes sufficiently alien and advanced enough that they appear to be magic to us.

**Comment by AdamElTablawy** [comment]
why are you talking about fucking realism this is a game where a black man in a maintenence shafts glares at you so hard a tumor explodes into your brain until a clown with a flare strapped to his dick hip thrusts it to death 

there are fucking wizards and slime cores that make fucking cameras and stop time

**Comment by KorPhaeron** [comment]
I thought it'd be obvious I was not serious considering I added half the magic shit in this game (sorry for ruining ss13 guys)

**Comment by optimumtact** [comment]
I got the joke

File: code/modules/food&drinks/drinks/drinks/bottle.dm [hunk 2]
```diff
@@ -389,3 +389,33 @@
 		user << "<span class='info'>You snuff out the flame on \the [src].</span>"
 		overlays -= fire_overlay
 		active = 0
+
+////////////////////////// POTIONS ///////////////////////
+
+/obj/item/weapon/reagent_containers/food/drinks/bottle/potion
+	name = "elixir flask"
+	desc = "My potions are too strong for you, traveler."
```
Comment (Niknakflak) [inline code/modules/food&drinks/drinks/drinks/bottle.dm]: I'm sorry, references are not allowed

Comment (PKPenguin321) [inline code/modules/food&drinks/drinks/drinks/bottle.dm]: nice meme

Comment (KorPhaeron) [inline code/modules/food&drinks/drinks/drinks/bottle.dm]: Should at least reference nethack or DnD or something instead

Comment (RemieRichards) [inline code/modules/food&drinks/drinks/drinks/bottle.dm]: https://www.youtube.com/watch?v=R_FQU4KzN7A

Comment (optimumtact) [inline code/modules/food&drinks/drinks/drinks/bottle.dm]: Niknak is wrong, references are allowed, or rather are up to the maintainers discretion.

The actual ban is on player references and is not that strongly enforced.

**Comment by coiax** [comment]
I AM INCAPABLE OF UNDERSTAND HUMOUR, AND FEEL THE DESIGN LEAD IS SERIOUS IN HIS COMPLAINT THIS PR ISN'T SCIENCY ENOUGH.

PLEASE ATTACH THE WORD "plasma" or "science" or "nuclear" or "bluespace" TO THE NAMES, THAT WILL RESOLVE ANY ISSUES.

**Comment by Zilenan91** [comment]
We could put these unmarked bottles in lavaland crates so that miners drink poison on accident.

**Comment by optimumtact** [comment]
We need a new buzzword for advanced science.

I propose

Quantum Nano Technology Tunneling

**Comment by Anonmare** [comment]
Something, something bluespace. Something, something sufficiently advanced technology.

**Comment by MMMiracles** [comment]
Is there a way to prevent pouring into other containers or otherwise cheesing the liquid out the bottle except for drinking it? 

Not really sure I see a way to do it that isn't super snowflakey as is.

**Comment by Jalleo** [comment]
Um isnt the transfering of the liquids a proc which can be overriden?

**Comment by MMMiracles** [comment]
If it is then I'd like to know it.

**Comment by Jalleo** [comment]
I believe that what you need to do is find where the defining of the inheritance for your bottles are for the transfering of the liquids and then essentially copy paste the check into a child of that attack proc while getting it so instead it prevents the transfer and inherit the rest of it. You should also check for the other things you shall need to override and make sure it works correctly.
@phil235 is this the right method?

File: code/modules/reagents/chemistry/reagents/potion_reagents.dm [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		else
+			if(show_message) //don't drink out of unlabeled flasks, kids
+				if(reac_volume > 10)
+					M << "<span class='userdanger'>You cry out in pain as you feel your insides become outsides!</span>"
+					M.visible_message("<span class='danger'><b>[M]</b> rapidly expands before exploding into blood confetti!")
+					M.gib()
+				else
+					M << "<span class='notice'>You burp, releasing some of the energy from the mixture.</span>"
+					M.emote("burp")
+		..()
```
Comment (phil235) [inline code/modules/reagents/chemistry/reagents/potion_reagents.dm]: no on_mob_life() for kinetic and healting potion?

Comment (optimumtact) [inline code/modules/reagents/chemistry/reagents/potion_reagents.dm]: they seem to just be using other reagents instead which is a bit weird

File: code/modules/reagents/chemistry/reagents/potion_reagents.dm [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	id = "genericpotion"
+	description = "A strange mixture of various components. There appears to be no real identifier to what its suppose to do."
+	color = "#FFF804" // rgb: 225, 248, 4
+	nutriment_factor = 0
+
+
+//healing potion
+/datum/reagent/consumable/potion/healing
+	id = "healingpotion"
+	description = "A strange mixture of various components. Pleasant fumes drift aimlessly off the mixture, just smelling it makes you feel better."
```
Comment (KorPhaeron) [inline code/modules/reagents/chemistry/reagents/potion_reagents.dm]: Using mixture two sentences in a row sounds off

**Comment by phil235** [comment]
>Is there a way to prevent pouring into other containers or otherwise cheesing the liquid out the bottle except for drinking it? 

YOu need to remove the OPENCONTAiNER flag or something.

**Comment by optimumtact** [comment]
Finish this or I'll finish you

**Comment by MMMiracles** [comment]
you cant finish me IF I FINISH MYSELF FIRST

**Pull request closed without merging**